### DISCUSSION TITLE
关于MVVM中的窗口切换

### DISCUSSION THREAD

**Discussion by xu112926** [BYJRK/DotNet-Discussions#81] [MVVM]
一些需要窗口切换的场景，比如
1.登录成功之后跳转主页面
2.点击按钮弹出用户输入窗口
在MVVM中如何处理比较好

**Comment by dumbnessrf** [comment]
同问

**Comment by BYJRK** [comment]
关于导航功能，可以参考[我的视频](https://www.bilibili.com/video/BV1a94y1r7cA)

关于弹出窗口，一般的思路无外乎这么几种：

- 使用原生的 MessageBox 等，只是将其抽象为一个接口，并实现相关的服务
- 使用 Popup
- 使用一个置于 MainWindow 上层的 ContentControl

**Reply by Timosthetic** [reply]
强烈推荐出一期视频，目前使用的窗体切换还停留在Prism  的showdialog
